src/wasm/wasm-validator.cpp

```diff
@@ -2903,6 +2903,9 @@ void FunctionValidator::visitFunction(Function* curr) {
                  "Multivalue function results (multivalue is not enabled)");
   }
   FeatureSet features;
+  // Check for things like having a rec group with GC enabled.
+  features |=
```

Comment:
Why are reference types ignored here?

Reply:
Because a function reference type always requires `ReferenceTypes` as a baseline, but it's kind of a false positive in this situation. MVP functions will have some MVP signature type here, but `getFeatures()` will still return `ReferenceTypes`.

In practice the only thing this line checks for is rec groups.

Reply:
Ah, right, the type itself is a reference type. A comment perhaps could help?

Reply:
Oh oops, I saw this comment just after merging. Will add a comment in a follow-up.